pkg/libs/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -55,6 +58,10 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 		return err
 	}
 
+	client, err := c.buildTLSClient()
```

Comment:
this is quite ineffective as we are building the HTTP client on every `sync` loop.
You don't want to reuse connections?

Reply:
I wouldn't reuse connections, shouldn't the point be that we want to make sure new connection can be created?

Reply:
but we might reuse clients to some degree, possibly caching on the TLS config?

Reply:
yeah, seems that the TLS config won't change very often